pynamodb/connection/base.py

```diff
@@ -281,20 +283,23 @@ def _make_api_call(self, operation_name, operation_kwargs):
             operation_kwargs,
             operation_model
         )
-
+        table_name = operation_kwargs.get(TABLE_NAME)
         prepared_request = self._create_prepared_request(request_dict, operation_model)
 
         for i in range(0, self._max_retry_attempts_exception + 1):
             attempt_number = i + 1
             is_last_attempt_for_exceptions = i == self._max_retry_attempts_exception
 
             try:
+                req_uuid = uuid.uuid4()
```

Comment:
IMHO we should move the hooks to surround linke 266 above so there is only one event fired even if there are retries and so we don't have to worry about a receiver throwing a `ValueError` and having pynamo interpret it as a failure it should retry.